src/endpoint/s3/ops/s3_post_object_select.js

```diff
@@ -0,0 +1,97 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const dbg = require('../../../util/debug_module')(__filename);
+const S3Error = require('../s3_errors').S3Error;
+const s3_utils = require('../s3_utils');
+const nb_native = require('../../../util/nb_native');
+const stream_utils = require('../../../util/stream_utils');
+
+/**
+ * https://docs.aws.amazon.com/AmazonS3/latest/API/API_SelectObjectContent.html
+ */
+async function post_object_select(req, res) {
+
+    if (nb_native().S3Select === undefined) {
+        dbg.error("Request for object select but compiled without BUILD_S3SELECT=1");
+        res.statusCode = 501;
+        res.end();
+        return;
```

Comment:
I wouldn't send just status code without a message - this will be very hard for users to figure out. 
What you should do here is throw an S3Error that will produce the correct code and message.